vividus/src/main/java/org/vividus/runner/AbstractGenericRunner.java

```diff
@@ -33,97 +27,61 @@
 import org.junit.runner.Result;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.vividus.BatchedEmbedder;
-import org.vividus.IBatchedPathFinder;
 import org.vividus.IRunStatusProvider;
 import org.vividus.Status;
 import org.vividus.configuration.BeanFactory;
 import org.vividus.configuration.Vividus;
 import org.vividus.log.TestInfoLogger;
 
-public class GenericRunner extends JUnitStories
+public abstract class AbstractGenericRunner extends JUnitStories
 {
     private static final int RUN_PASSED_EXIT_CODE = 0;
     private static final int RUN_KNOWN_ISSUES_EXIT_CODE = 1;
     private static final int RUN_FAILED_EXIT_CODE = 2;
     private static final int ERROR_EXIT_CODE = 3;
 
-    private static String embedderBeanName;
-    private final Embedder embedder;
-    private final IBatchedPathFinder batchedPathFinder;
+    private static Class<?> runnerClass;
 
-    public GenericRunner()
+    public AbstractGenericRunner()
     {
         Vividus.init();
-        batchedPathFinder = BeanFactory.getBean(IBatchedPathFinder.class);
-        embedder = BeanFactory.getBean(embedderBeanName, Embedder.class);
         TestInfoLogger.logPropertiesSecurely(System.getProperties());
         TestInfoLogger.logPropertiesSecurely(BeanFactory.getBean("properties", Properties.class));
     }
 
-    protected static void setEmbedderBeanName(String embedderBeanName)
+    protected static void setRunnerClass(Class<?> runnerClass)
     {
-        GenericRunner.embedderBeanName = embedderBeanName;
+        AbstractGenericRunner.runnerClass = runnerClass;
     }
 
     @Override
     public Embedder configuredEmbedder()
     {
-        return embedder;
+        return getEmbedder();
     }
 
     @Override
     public Configuration configuration()
     {
-        return embedder.configuration();
+        return getEmbedder().configuration();
     }
 
     @Override
     public InjectableStepsFactory stepsFactory()
     {
-        return embedder.stepsFactory();
+        return getEmbedder().stepsFactory();
     }
 
-    @Override
-    public List<String> storyPaths()
-    {
-        return getPaths().values().stream().flatMap(List::stream).collect(Collectors.toList());
-    }
-
-    @Override
-    public void run()
-    {
-        Embedder embedder = configuredEmbedder();
-        if (embedder instanceof BatchedEmbedder)
-        {
-            ((BatchedEmbedder) embedder).runStoriesAsPaths(getPaths());
-        }
-        else
-        {
-            super.run();
-        }
-    }
-
-    private Map<String, List<String>> getPaths()
-    {
-        try
-        {
-            return batchedPathFinder.getPaths();
-        }
-        catch (IOException e)
-        {
-            throw new UncheckedIOException(e);
-        }
-    }
+    protected abstract Embedder getEmbedder();
```

Comment:
just use `configuredEmbedder()`

Reply:
okay, but I added getEmbedder because in the parent class configuredEmbedder() is not required to override, so we may forget it in the future